### DISCUSSION TITLE
Issues with newer Linux distros

### DISCUSSION THREAD

**Discussion by axelrafn** [troglobit/smcroute#197] [Q&A] [answered]
Hi and thank you for the software.

We are running it to relay radio stations to endpoints within our VSAT network and older versions have worked fine without any problems for years. But lately, we've been setting up newer distributions of Linux on newer hardware to upgrade older machines and we seem to have hit a brick wall.

The thing is that we get everything running with our old script, but it only routes the multicast traffic for a short while until it stops. It always stops just under 5000 packets or around 6.4-6.7MB.

We've gone over the documentation and done what we can to debug this but still come up empty.
I've converted the old shell script we've had to the newer /etc/smcroute.conf and it is working but the routing still stops at those ~5000 packets.

Is there something anyone knows of that might help us in this situation?

**Comment by troglobit** [comment]
Hi, always a pleasure hearing back from long-time users of SMCRoute! :smiley: 

Regarding your problem, can't say I've ever seen anything with those symptoms before. But tell me a bit more about _your setup_ and the _exact version of SMCRoute_ you are using.  E.g., is it running on top of a _Linux bridge_ or something else worth mentioning, and in relation to the _setup/topology_, how does _your `smcroute.conf` look_?

**Comment by axelrafn** [comment]
Sure thing.

First off, the machine I'm working on right now is a Debian stable (Bullseye  11.x) running in a VM on a Proxmox host (with Linux bridges yes). The Debian stable is running smcroute 2.4.4, but this also happens on a standalone machine (no Proxmox) which is a Debian 8.4 running smcroute 2.4.2.

The script that I mentioned was made well before I started working here and it has the following:

> #!/bin/bash
> sleep 2
> echo "restarting smcroute if running"
> /usr/sbin/smcroute -k
> sleep 1
> /usr/sbin/smcroute -d
> sleep 2
> echo "adding route for multicast"
> /usr/sbin/smcroute -a ens18 10.10.12.249 228.5.6.7 ens20
> /usr/sbin/smcroute -a ens18 10.10.12.237 228.5.6.8 ens20
> /usr/sbin/smcroute -a ens18 10.10.12.239 228.5.6.11 ens20
> #sleep 2
> echo "joining group"
> /usr/sbin/smcroute -j ens18 228.5.6.7
> /usr/sbin/smcroute -j ens18 228.5.6.8
> /usr/sbin/smcroute -j ens18 228.5.6.11
> sleep 3
> echo "done"

The /etc/smcroute.conf that I created based on the above script and used on the hosted machine contains:

> phyint ens18
> phyint ens20
> 
> mgroup from ens18 group 228.5.6.7
> mgroup from ens18 group 228.5.6.7 source 10.10.12.249
> mroute from ens18 group 228.5.6.7 source 10.10.12.249 to ens20
> 
> mgroup from ens18 group 228.5.6.8
> mgroup from ens18 group 228.5.6.8 source 10.10.12.237
> mroute from ens18 group 228.5.6.8 source 10.10.12.237 to ens20
> 
> mgroup from ens18 group 228.5.6.11
> mgroup from ens18 group 228.5.6.11 source 10.10.12.239
> mroute from ens18 group 228.5.6.11 source 10.10.12.239 to ens20`

To explain how the network looks like, we have a computer at ground zero that is pumping out via multicast, audio streams. Those multicast streams then travel over VSAT connections to the endpoints where computers (the ones above) relay the multicast streams to the user base behind them.

Older, non-maintained machines (most running Ubuntu 16.04 or older) have this working without a flaw, so we know the multicast streams are reaching each client cell. But on newer machines and newer operating systems, it seems to just stop at ~4500-4900 packets. I've been playing around with /etc/sysctl.conf to tweak some kernel settings as per many posts on the Internet but without success.

That's why I was hoping you could perhaps give some insight into what I'm doing wrong.
I also have to mention that this is my first time ever working on multicast. Never touched it before I started here about 9 months ago.

Hope this answers your questions.

**Comment by axelrafn** [comment, marked as answer]
I think I stumbled upon the error myself, it seems that the mangle rule in iptables got wiped out. After I manually set it up again everything works fine now. =D

**Reply by troglobit** [reply]
Sorry for the late reply, but great to hear you found the root cause yourself! :-)
